gossip/src/cluster_info.rs

```diff
@@ -2527,8 +2527,8 @@ impl ClusterInfo {
             should_check_duplicate_instance,
         )?;
         if last_print.elapsed() > SUBMIT_GOSSIP_STATS_INTERVAL {
-            submit_gossip_stats(&self.stats, &self.gossip, &stakes);
-            *last_print = Instant::now();
+            // submit_gossip_stats(&self.stats, &self.gossip, &stakes);
+            // *last_print = Instant::now();
```

Comment:
So ”submit” actually resets the counters?

I can think of two ways to make this work upstream.

1. Make logging metrics and Prometheus metrics mutually exclusive, and put the choice in a CLI flag. Then if Prometheus metrics are enabled, we would skip the log + reset here.
2. If `GossipStats` holds only counters, we could implement `std::ops::Sub` for it, and keep two instances of it: the current stats, and the last logged stats. Then when it is time to log, we log `current - last_logged`, and then set `last_logged = current`. That way the metrics remain increasing for Prometheus, and the log output remains unchanged.

I think option 2 is kind of neat. It’s a bit more tedious in the `GossipStats`, but it may be simpler in the end than threading a CLI flag through everything to this point. And it’s nicer to not have to choose.